assets/src/components/detours/detourMap.tsx

```diff
@@ -1,152 +1,120 @@
+   */
+  startPoint?: LatLngLiteral
+
+  /**
+   * Coordinate to display as the ending connection point.
+   */
+  endPoint?: LatLngLiteral
+
+  /**
+   * Coordinates to display as the waypoints.
+   */
+  waypoints: LatLngLiteral[]
+
+  /**
+   * Callback fired when the {@link originalShape} is clicked.
+   */
+  onClickOriginalShape: (point: LatLngLiteral) => void
+
+  /**
+   * Callback fired when the map is clicked.
+   * @param point
+   */
+  onClickMap: (point: LatLngLiteral) => void
+
+  /**
+   * User signal to describe the state of the undo button.
+   */
+  undoDisabled: boolean
+  /**
+   * Callback fired when the undo button is clicked.
+   */
+  onUndoLastWaypoint: () => void
 }
 
-const RouteShapeWithDetour = ({
+export const DetourMap = ({
   originalShape,
+  detourShape,
+
   startPoint,
-  onSetStartPoint,
   endPoint,
-  onSetEndPoint,
   waypoints,
-  onAddDetourPosition,
-  detourShapePositions,
-}: {
-  originalShape: Shape
-  startPoint: LatLngLiteral | null
-  onSetStartPoint: (p: LatLngLiteral | null) => void
-  endPoint: LatLngLiteral | null
-  onSetEndPoint: (p: LatLngLiteral | null) => void
-  waypoints: LatLngLiteral[]
-  onAddDetourPosition: (p: LatLngLiteral) => void
-  detourShapePositions: LatLngLiteral[]
-}) => {
-  const routeShapePositions: LatLngLiteral[] = originalShape.points.map(
-    shapePointToLatLngLiteral
-  )
-
-  useMapEvent("click", (e) => {
-    if (startPoint !== null && endPoint === null) {
-      onAddDetourPosition(e.latlng)
-    }
-  })
-
-  // points on the detour not already represented by the start and end
-  const uniqueWaypoints =
-    waypoints.length === 0
-      ? []
-      : endPoint === null
-      ? waypoints.slice(1)
-      : waypoints.slice(1, -1)
-
-  return (
-    <>
-      <Polyline
-        positions={routeShapePositions}
-        className="c-detour_map--original-route-shape"
-        eventHandlers={{
-          click: (e) => {
-            if (startPoint === null) {
-              const { position } =
-                closestPosition(routeShapePositions, e.latlng) ?? {}
-
-              position && onSetStartPoint(position)
-              position && onAddDetourPosition(position)
-            } else if (endPoint === null) {
-              const { position } =
-                closestPosition(routeShapePositions, e.latlng) ?? {}
-
-              position && onSetEndPoint(position)
-              position && onAddDetourPosition(position)
-            }
-          },
-        }}
-        bubblingMouseEvents={false}
-      />
-      {startPoint && <StartMarker position={startPoint} />}
-      {endPoint && <EndMarker position={endPoint} />}
-      <Polyline
-        positions={detourShapePositions}
-        className="c-detour_map--detour-route-shape"
-      />
-      {uniqueWaypoints.map((position) => (
-        <DetourPointMarker key={position.toString()} position={position} />
-      ))}
-    </>
-  )
+
+  onClickOriginalShape: onAddConnectionPoint,
```

Comment:
Nit - should we stick with either `onClickOriginalShape` or `clickOriginalShape` for this, rather than `addConnectionPoint`? Given this refactor, it feels like the fact that this operation adds a connection point is a diversion-page fact, not a detour-map fact.

Reply:
Whoops! that was unintentional, sorry about that.
See 382b93547fa632bf40e5d5862965ba62eb99acd8

Reply:
This was a side-effect of using VSCode to rename the symbols and forgetting to clean up after it 🫠